src/Compilers/Core/Portable/CodeGen/PrivateImplementationDetails.cs

```diff
@@ -590,46 +687,146 @@ public override string ToString()
+        public override IEnumerable<Cci.IFieldDefinition> GetFields(EmitContext context) => _fields;
+        public override IEnumerable<Cci.IMethodDefinition> GetMethods(EmitContext context) => _methods;
+        public override bool IsBeforeFieldInit => true;
+
+        private sealed class DataSectionStringField(
+            string name, Cci.INamedTypeDefinition containingType, PrivateImplementationDetails privateImplDetails)
+            : SynthesizedStaticFieldBase(name, containingType)
+        {
+            private readonly PrivateImplementationDetails _privateImplDetails = privateImplDetails;
+
+            public override ImmutableArray<byte> MappedData => default;
+            public override bool IsReadOnly => true;
+
+            public override Cci.ITypeReference GetType(EmitContext context)
+            {
+                return _privateImplDetails.ModuleBuilder.GetPlatformType(Cci.PlatformType.SystemString, context);
+            }
+
+            public override string ToString()
+            {
+                return $"string {(object?)ContainingTypeDefinition.GetInternalSymbol() ?? ContainingTypeDefinition}.{this.Name}";
+            }
+        }
     }
 
-    internal abstract class SynthesizedStaticField : Cci.IFieldDefinition
+    internal abstract class SynthesizedStaticField : SynthesizedStaticFieldBase
     {
-        private readonly Cci.INamedTypeDefinition _containingType;
         private readonly Cci.ITypeReference _type;
-        private readonly string _name;
 
         internal SynthesizedStaticField(string name, Cci.INamedTypeDefinition containingType, Cci.ITypeReference type)
+            : base(name, containingType)
+        {
+            RoslynDebug.Assert(type != null);
+
+            _type = type;
+        }
+
+        internal Cci.ITypeReference Type => _type;
+
+        public override string ToString() => $"{(object?)_type.GetInternalSymbol() ?? _type} {(object?)ContainingTypeDefinition.GetInternalSymbol() ?? ContainingTypeDefinition}.{this.Name}";
+
+        public override Cci.ITypeReference GetType(EmitContext context) => _type;
+    }
+
+    internal abstract class SynthesizedStaticFieldBase : Cci.IFieldDefinition
+    {
+        private readonly Cci.INamedTypeDefinition _containingType;
+        private readonly string _name;
+
+        internal SynthesizedStaticFieldBase(string name, Cci.INamedTypeDefinition containingType)
         {
             RoslynDebug.Assert(name != null);
             RoslynDebug.Assert(containingType != null);
-            RoslynDebug.Assert(type != null);
 
             _containingType = containingType;
-            _type = type;
             _name = name;
         }
 
-        public override string ToString() => $"{(object?)_type.GetInternalSymbol() ?? _type} {(object?)_containingType.GetInternalSymbol() ?? _containingType}.{this.Name}";
+        public abstract override string ToString();
```

Comment:
>public abstract override string ToString(); [](http://example.com/codeflow?start=8&length=43)

It feels unnecessary to require a derived type to override `ToString`. It is, obviously, a choice to override the method, but `ToString` behavior is not important for the purpose of the type.

Reply:
It is displayed in IL snapshots in tests, so it is kind of important, but I can see that's perhaps not for the purpose of the type, will remove.